File: modulefederation/apps/web/src/app/workflow/service/process-definition-service.ts
import { Injectable, Injector } from "@angular/core";
import { AbstractHTTPService, Service } from "@modulefederation/portal";
import { Observable } from 'rxjs';


 export interface PropertyDescriptor {
  name: string,
  type: string,
  constraint: string,
  value?: any
 }

 export interface SchemaDescriptor {
  properties: PropertyDescriptor[]
 }

 export interface TaskSchema {
    process: SchemaDescriptor,
    input: SchemaDescriptor,
    output: SchemaDescriptor,
}
  

 @Injectable({providedIn: 'root'})
 @Service({domain: "admin", prefix: "/process-definition/"})
 export class ProcessDefinitionService extends AbstractHTTPService {
     // constructor
 
     constructor(injector : Injector) {
         super(injector);
     }
 
     // methods


    processSchema(process: String, deployment: number) : Observable<SchemaDescriptor> {
        return this.get<SchemaDescriptor>(`process-schema/${process}/${deployment}`)
    }

    taskInputSchema(process: String, deployment: number, task: String) : Observable<SchemaDescriptor> {
      return this.get<SchemaDescriptor>(`task-input-schema/${process}/${deployment}/${task}`)
    }

    taskSchema(process: String, deployment: number, task: String) : Observable<TaskSchema> {
        return this.get<TaskSchema>(`task-schema/${process}/${deployment}/${task}`)
      }
 } 

File: workflow/core/src/main/java/org/sirius/workflow/service/ProcessDefinitionService.kt
package org.sirius.workflow.service
/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

import org.springframework.web.bind.annotation.*

/*data class PropertyDescriptor(val name: String, val type: String, var value: Any?)

data class SchemaDescriptor(val properties: List<PropertyDescriptor>)
*/
data class TaskSchema(val process: SchemaDescriptor, val input: SchemaDescriptor, val output: SchemaDescriptor)

//@ServiceInterface
@CrossOrigin(
    origins = [ "http://localhost:8080", "http://localhost:4200"],
    methods = [
        RequestMethod.OPTIONS,
        RequestMethod.GET,
        RequestMethod.PUT,
        RequestMethod.DELETE,
        RequestMethod.POST
    ])
@RequestMapping("process-definition/")
@RestController
interface ProcessDefinitionService /*: Service*/ {
    @GetMapping("process-schema/{process}/{deployment}")
    @ResponseBody
    fun processSchema(@PathVariable process: String,  @PathVariable deployment: Long) : SchemaDescriptor

    @GetMapping("task-schema/{process}/{deployment}/{task}")
    @ResponseBody
    fun taskSchema(@PathVariable process: String,  @PathVariable deployment: Long, @PathVariable task: String) : TaskSchema

    @GetMapping("task-input/{process}/{deployment}/{task}")
    @ResponseBody
    fun taskInputSchema(@PathVariable process: String,  @PathVariable deployment: Long, @PathVariable task: String) : SchemaDescriptor
}
